lib/mongoid/association/relatable.rb

```diff
@@ -411,7 +411,7 @@ def default_inverse
       def namespace_hierarchy(mod)
         parent = Object
         hier = [parent]
-        mod.name.split('::').each do |part|
+        mod.name&.split('::')&.each do |part|
```

Comment:
@Neilshweky this logic breaks with anonymous classes (e.g. as used in specs.) Adding this safe navigation `&.` check seems to resolve the issue.